includes/class-wporg-gp-translation-events-stats-calculator.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+class WPORG_GP_Translation_Events_Stats_Row {
+	public int $created;
+	public int $reviewed;
+	public int $users;
+
+	public function __construct( $created, $reviewed, $users ) {
+		$this->created  = $created;
+		$this->reviewed = $reviewed;
+		$this->users    = $users;
+	}
+}
+
+class WPORG_GP_Translation_Events_Event_Stats {
+	/**
+	 * Associative array of rows, with the locale as key.
+	 *
+	 * @var WPORG_GP_Translation_Events_Stats_Row[]
+	 */
+	private array $rows = [];
+
+	private WPORG_GP_Translation_Events_Stats_Row $totals;
+
+	/**
+	 * @throws Exception
+	 */
+	public function add_row( string $locale, WPORG_GP_Translation_Events_Stats_Row $row ) {
+		if ( ! $locale ) {
+			throw new Exception( 'locale must not be empty' );
+		}
+		$this->rows[ $locale ] = $row;
+	}
+
+	public function set_totals( WPORG_GP_Translation_Events_Stats_Row $totals ) {
+		$this->totals = $totals;
+	}
+
+	/**
+	 * Get an associative array of rows, with the locale as key.
+	 *
+	 * @return WPORG_GP_Translation_Events_Stats_Row[]
+	 */
+	public function rows(): array {
+		return $this->rows;
+	}
+
+	public function totals(): WPORG_GP_Translation_Events_Stats_Row {
+		return $this->totals;
+	}
+}
+
+class WPORG_GP_Translation_Events_Stats_Calculator {
+	private string $ACTIONS_TABLE_NAME = WPORG_GP_Translation_Events_Translation_Listener::ACTIONS_TABLE_NAME;
+
+	/**
+	 * @throws Exception
+	 */
+	function for_event( WP_Post $event ): WPORG_GP_Translation_Events_Event_Stats {
+		$start = new DateTime( get_post_meta( $event->ID, '_event_start', true ), new DateTimeZone( 'UTC' ) );
+		$end   = new DateTime( get_post_meta( $event->ID, '_event_end', true ), new DateTimeZone( 'UTC' ) );
+		$stats = new WPORG_GP_Translation_Events_Event_Stats;
+		global $wpdb;
+
+		$query = $wpdb->prepare( "
+				select locale,
+					   sum(if(action = 'create', 1, 0)) AS created,
+					   sum(if(action in ('approve', 'reject', 'request_changes', 'mark_fuzzy'), 1, 0)) AS reviewed,
```

Comment:
```suggestion
					   sum(action = 'create') AS created,
					   count(*) AS total,
```

Can we make this simpler and calculate the reviewed as `total - created` below?

Reply:
[MySQL's SUM function](https://dev.mysql.com/doc/refman/8.0/en/functions.html#function_sum) is defined as `[SUM([DISTINCT] expr) [over_clause]` so you don't need the if.

Reply:
Makes sense!